### PR TITLE
Feature/docs 2251 credential stuffing

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
 
-[Credential stuffing](https://owasp.org/www-community/attacks/Credential_stuffing) is the automated injection of stolen or weak username/email and password pairs (credentials) into website login forms, in order to fraudulently gain access to user accounts. This article describes how to detect this type of threats using Wallarm's **Credential Stuffing Detection**.
-
-![Wallarm Console - Credential Stuffing](../images/about-wallarm-waf/credential-stuffing/credential-stuffing.png)
+[Credential stuffing](../attacks-vulns-list.md#credential-stuffing) is a cyber attack where hackers use lists of compromised user credentials to gain unauthorized access to user accounts on multiple websites. This attack is hazardous because many people reuse the same username and password across different services or use popular weak passwords. A successful credential stuffing attack requires fewer attempts, so attackers can send requests much less frequently, which makes standard measures like brute force protection ineffective. This article describes how to detect this type of threats using Wallarm's **Credential Stuffing Detection**.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:3]: 1. fewer than what?
2. for intro, this text is too big

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
 
 ## How Wallarm addresses credential stuffing
 
-Wallarm's **Credential Stuffing Detection** collects and displays real-time information about compromised and weak credentials in your application, as well as attempts to use them. It also enables instant notifications about such attempts.
+Wallarm's **Credential Stuffing Detection** collects and displays real-time information about attempts to use compromised or weak credentials to access your applications. It also enables instant notifications about such attempts and forms full downloadable list of all compromised or weak credentials providing access to your applications.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:7]: full is extra, you already said "all compromised ..."

File: docs/latest/quickstart/attack-prevention-best-practices.md [hunk 3]
```diff
@@ -50,6 +50,10 @@ The Broken Object Level Authorization (BOLA) vulnerability allows an attacker to
 
 [Set up API abuse profiles](../user-guides/api-abuse-prevention.md) to stop and block bots performing API abuse like account takeover, scraping, security crawlers and other automated malicious actions targeted at your APIs.
 
+## Enable credential stuffing detection
+
+Enable [credential stuffing detection](../about-wallarm/credential-stuffing.md) to have a real-time information about attempts to use compromised or weak credentials to access your applications and a full downloadable list of all compromised or weak credentials providing access to your applications.
+
 ## Create custom attack detection rules
```
Comment (AnastasiaTWW) [inline docs/latest/quickstart/attack-prevention-best-practices.md:58]: is the current [credential-stuffing-schema.png](https://github.com/wallarm/product-documentation/pull/1092/files#diff-4220766845dd66c41a1d79546577fcb7d007b8ff77e70783f90345ea1fdf944e) a draft? there are different fonts, wrong location of texts in comparison to our standard schemes, extra capitalization

Comment (AnastasiaTWW) [reply docs/latest/quickstart/attack-prevention-best-practices.md:58]: application icon is not named but the text of the arrow going to this ison says "applicatrion"

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+
+To identify compromised and weak passwords, Wallarm uses a comprehensive database of more than **850 million records** collected from the public [HIBP](https://haveibeenpwned.com/) compromised credentials database.
+
+![Credential Stuffing - Schema](../images/about-wallarm-waf/credential-stuffing/credential-stuffing-schema.png)
+
+Wallarm's Credential Stuffing Detection keeps credentials data safe applying the following sequence of actions:
+
+1. As request arrives to the node, it counts [SHA-1](https://en.wikipedia.org/wiki/SHA-1) from the password and sends several chars to the Cloud.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:15]: missed articles in this block of the information

Comment (AnastasiaTWW) [reply docs/latest/about-wallarm/credential-stuffing.md:15]: counts is not a correct term. calculate, generate, compute but not count as it is not counting items

Comment (AnastasiaTWW) [reply docs/latest/about-wallarm/credential-stuffing.md:15]: 3. from what parameters of requests we get passwords?

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
+1. Cloud sends full SHA-1 encrypted compromised passwords that begin with the received chars back to the node.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:16]: not really clear how it works. am I right that on the CLoud side we check our database of known compromised passwords looking for those starting with the received chars? and if we find it, we send this password in the SHA-1 encrypted format to the node for it to compare the password from the request with the ones received from the Cloud? if it is a match, we consider that the request uses a compormised account data and record this request correspondingly putting not only password to credential information but also linked login/email? and then we pass the request to the application. if so, then the current list of items does not include all the information and must be fixed

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 6]
```diff
@@ -1,14 +1,29 @@
...
+1. If there is a match, the node reports a credential stuffing attack to the Cloud, the login taken from the request is included into this attack information.
+1. The node passes request to the application.
+
+Thus, credentials are never sent together, passwords are never sent unencrypted, and your clients' authorization data always stays secure.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:20]: sent where to?

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 7]
```diff
@@ -1,14 +1,29 @@
...
+
+**Mass and single attempts**
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:22]: why not to put it as a header?

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 8]
```diff
@@ -1,14 +1,29 @@
...
+
+Credential Stuffing Detection is capable of registering both massive attempts of usage of the compromised credentials performed by bots and single attempts, undetectable by other means.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:24]: not clear where "both" relates to

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 9]
```diff
@@ -34,24 +47,26 @@ Once **Credential Stuffing Detection** is enabled, a [configuration](#configurin
...
 There are two ways of adding endpoints to the list:
 
 * From the **Recommended endpoints** list that includes two types of elements:
 
     * Wallarm's predefined rules utilizing regular expressions to specify commonly used authentication endpoints and their parameters storing passwords and logins.
-
+    <!--
         ![Credential Stuffing - Recommended Endpoints - Predefined rules](../images/about-wallarm-waf/credential-stuffing/credential-stuffing-predefined-rules.png)
-
-    * Endpoints used for authentication that were found by the [API Discovery](../api-discovery/overview.md) module and recorded as they actually received traffic. This is available only if API Discovery is presented in your subscription and enabled.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:45]: why is it commented out?

File: docs/latest/attacks-vulns-list.md [hunk 10]
```diff
@@ -812,6 +809,32 @@ You may follow these recommendations:
...
 
+### Credential stuffing
+
+**Attack**
+
+**Wallarm code:** `credential_stuffing`
+
+**Description:**
+
+A cyber attack where hackers use lists of compromised user credentials to gain unauthorized access to user accounts on multiple websites. This attack is hazardous because many people reuse the same username and password across different services or use popular weak passwords. A successful credential stuffing attack requires fewer attempts, so attackers can send requests much less frequently, which makes standard measures like brute force protection ineffective.
```
Comment (AnastasiaTWW) [inline docs/latest/attacks-vulns-list.md:820]: how do user credentials become compromiused? I think, it is worthy to shortly explain it here

Comment (AnastasiaTWW) [reply docs/latest/attacks-vulns-list.md:820]: I would avoid phrases like "people often" as it is too direct for technical content

the common practice of reusing identical usernames and passwords across different services, along with the tendency to choose easily guessable passwords.

File: docs/latest/quickstart/attack-prevention-best-practices.md [hunk 11]
```diff
@@ -50,6 +50,10 @@ The Broken Object Level Authorization (BOLA) vulnerability allows an attacker to
 
 [Set up API abuse profiles](../user-guides/api-abuse-prevention.md) to stop and block bots performing API abuse like account takeover, scraping, security crawlers and other automated malicious actions targeted at your APIs.
 
+## Enable credential stuffing detection
+
+Enable [credential stuffing detection](../about-wallarm/credential-stuffing.md) to have a real-time information about attempts to use compromised or weak credentials to access your applications and a full downloadable list of all compromised or weak credentials providing access to your applications.
```
Comment (AnastasiaTWW) [inline docs/latest/quickstart/attack-prevention-best-practices.md:55]: and what to do with this? the value is not clear

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 1 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:3]: 1. "fewer than what?" - brute force is mentioned later in the same sentence.
2. fixed

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 4 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:15]: > from what parameters of requests we get passwords?
That will be explained in Configuring section. This intro should stay short and focused on safety, not on the technical details.

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 5 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:16]: Correct up to step 3 - the node creates credential stuffing attack here and includes **only login**. With this correction everything is correct and it is exactly what the text says now.

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 6 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:20]: It is on the diagram. Sent back and force between node and the Cloud.

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 7 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:22]: That would be too much of a highlight.

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 8 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:24]: "both massive and single" + it is highlighted in the header.

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 9 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:45]: By the request from Alexander to reduce the number of screenshots in this section. I did not delete them completely as I feel we can change our minds and the screenshots themselves contain a valuable information.

File: docs/latest/about-wallarm/credential-stuffing.md [hunk 12]
```diff
@@ -1,21 +1,25 @@
...
 
-[Credential stuffing](../attacks-vulns-list.md#credential-stuffing) is a cyber attack where hackers use lists of compromised user credentials to gain unauthorized access to user accounts on multiple websites. This attack is hazardous because many people reuse the same username and password across different services or use popular weak passwords. A successful credential stuffing attack requires fewer attempts, so attackers can send requests much less frequently, which makes standard measures like brute force protection ineffective. This article describes how to detect this type of threats using Wallarm's **Credential Stuffing Detection**.
+[Credential stuffing](../attacks-vulns-list.md#credential-stuffing) is a cyber attack where hackers use lists of compromised user credentials to gain unauthorized access to user accounts on multiple websites. This article describes how to detect this type of threats using Wallarm's **Credential Stuffing Detection**.
+
+User credentials can become compromised in different ways, including as a result of email fishing, social engineering, disclosure by insiders or technical data breach.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/credential-stuffing.md:5]: delete

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 5 above)
Comment (AnastasiaTWW) [reply docs/latest/about-wallarm/credential-stuffing.md:20]: I'm struggling with the 2nd step as it is not clear where does the Cloud get "SHA-1 encrypted compromised passwords that begin with the received chars back to the node."
as for the 3rd step, "a match" is not clear wjhat is it - what is compared

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 6 above)
Comment (AnastasiaTWW) [reply docs/latest/about-wallarm/credential-stuffing.md:24]: the fact that the diagram says where the data is sent to does not mean that the text may stay incomplete.

Thus, passwords from machines with Wallarm nodes are never sent to the Wallarm Cloud unencrypted. Credentials are not sent simultaneously, ensuring clients' authorization data remains secure within your network.

File: docs/latest/quickstart/attack-prevention-best-practices.md [hunk 13]
```diff
@@ -52,7 +52,9 @@ The Broken Object Level Authorization (BOLA) vulnerability allows an attacker to
 
 ## Enable credential stuffing detection
 
-Enable [credential stuffing detection](../about-wallarm/credential-stuffing.md) to have a real-time information about attempts to use compromised or weak credentials to access your applications and a full downloadable list of all compromised or weak credentials providing access to your applications.
+Enable [credential stuffing detection](../about-wallarm/credential-stuffing.md) to have a real-time information about attempts to use compromised or weak credentials to access your applications and a downloadable list of all compromised or weak credentials providing access to your applications.
+
+Knowledge of accounts with stolen or weak passwords allows you to initiate measures to secure these accounts' data, like communicating with account owners, temporarily suspending access to the accounts, etc.
 
 ## Create custom attack detection rules
```
Comment (AnastasiaTWW) [inline docs/latest/quickstart/attack-prevention-best-practices.md:60]: the scheme must be fixed

File: docs/latest/about-wallarm/credential-stuffing.md (hunk 5 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/credential-stuffing.md:20]: Fixed

**Pull request merged**